application/src/main/java/org/thingsboard/server/service/gateway_device/DefaultGatewayDeviceStateService.java

```diff
@@ -0,0 +1,106 @@
+import org.thingsboard.server.common.data.rpc.ToDeviceRpcRequestBody;
+import org.thingsboard.server.common.msg.rpc.ToDeviceRpcRequest;
+import org.thingsboard.server.dao.device.DeviceService;
+import org.thingsboard.server.service.rpc.TbCoreDeviceRpcService;
+
+import java.util.UUID;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class DefaultGatewayDeviceStateService implements GatewayDeviceStateService {
+
+    private final static String DEVICE_RENAMED_METHOD_NAME = "gateway_device_renamed";
+    private final static String DEVICE_DELETED_METHOD_NAME = "gateway_device_deleted";
+    private final DeviceService deviceService;
+    @Value("${server.rest.server_side_rpc.min_timeout:5000}")
+    protected long minTimeout;
+    @Value("${server.rest.server_side_rpc.default_timeout:10000}")
+    protected long defaultTimeout;
+    @Lazy
+    @Autowired
+    private TbCoreDeviceRpcService deviceRpcService;
+
+    @Override
+    public void update(Device device, Device oldDevice) {
+        Device gatewayDevice = findGatewayDeviceByAdditionalInfoInDevice(device.getTenantId(), device.getAdditionalInfo());
+        if (gatewayDevice != null) {
+            ObjectNode renamedDeviceNode = JacksonUtil.newObjectNode();
+            renamedDeviceNode.put(oldDevice.getName(), device.getName());
+            ToDeviceRpcRequest rpcRequest = formDeviceToGatewayRPCRequest(gatewayDevice, renamedDeviceNode, DEVICE_RENAMED_METHOD_NAME);
+            deviceRpcService.processRestApiRpcRequest(rpcRequest, fromDeviceRpcResponse -> {
+                log.trace("Device renamed RPC with id: [{}] processed to gateway device with id: [{}], old device name: [{}], new device name: [{}]",
+                        rpcRequest.getId(), gatewayDevice.getId(), oldDevice.getName(), device.getName());
+            }, null);
+        }
+    }
+
+    @Override
+    public void delete(Device device) {
+        Device gatewayDevice = findGatewayDeviceByAdditionalInfoInDevice(device.getTenantId(), device.getAdditionalInfo());
+        if (gatewayDevice != null) {
+            TextNode deletedDeviceNode = new TextNode(device.getName());
+            ToDeviceRpcRequest rpcRequest = formDeviceToGatewayRPCRequest(gatewayDevice, deletedDeviceNode, DEVICE_DELETED_METHOD_NAME);
+            deviceRpcService.processRestApiRpcRequest(rpcRequest, fromDeviceRpcResponse -> {
+                log.trace("Device deleted RPC with id: [{}] processed to gateway device with id: [{}], deleted device name: [{}]",
+                        rpcRequest.getId(), gatewayDevice.getId(), device.getName());
+            }, null);
+        }
+    }
+
+    private ToDeviceRpcRequest formDeviceToGatewayRPCRequest(Device gatewayDevice, JsonNode deviceDataNode, String method) {
+        ToDeviceRpcRequestBody body = new ToDeviceRpcRequestBody(method, JacksonUtil.toString(deviceDataNode));
+        long expTime = System.currentTimeMillis() + Math.max(minTimeout, defaultTimeout);
```

Comment:
Please explain this logic: Math.max(minTimeout, defaultTimeout)? Both parameters are from yml file, right? And they are constant. Why not to calculate this in the init() method? I believe that the timeout value of 10 seconds is too low for the persistent RPC query. Put 1 day or similar.